homeassistant/components/cover/tahoma.py

```diff
@@ -33,7 +33,13 @@ class TahomaCover(TahomaDevice, CoverDevice):
     def __init__(self, tahoma_device, controller):
         """Initialize the Tahoma device."""
         super().__init__(tahoma_device, controller)
-        self.entity_id = ENTITY_ID_FORMAT.format(self.unique_id)
+        eid = ENTITY_ID_FORMAT.format(self.unique_id)
+        # strip off the RTS or the IO ID from the entity ID
+        if eid.rfind('_rts') > 0:
+            eid = eid[:eid.rfind('_rts')]
+        elif eid.rfind('_io') > 0:
+            eid = eid[:eid.rfind('_io')]
+        self.entity_id = eid
```

Comment:
Do you realy need to set the entity_id and don't let's auto generate from internal core/name ?

Reply:
The point is that - as mentioned above - the current entity_id looks like ```<custom_name>_rts_<NNN>```, where ```<NNN>``` is a very long numeric ID internally used by the TaHoma box to reference the shutters. This makes writing automations pretty unpractical (and the produced code unreadable).

IMHO Home Assistant should abstract from such platform-dependent details and expose high-level identifiers, so I believe it is definitely worth to override the auto-generated identifiers here.

Reply:
@pvizeli Currently an entity id look like this: ```cover.bedroom_middle_io12345678901234567890``` . Having ```cover.bedroom_middle``` would be more desireable, right?

Reply:
Yes. Home-Assistant generate automatic entity_id with name. Remove the entity_id and the core do that automatic. For unique_id, you can only use the `NN` if they is unique.

Reply:
So the question is about the pre-existing statement at [L36](https://github.com/home-assistant/home-assistant/pull/11547/files/5a86b1986610cf02d45f09f9d11ee95e2c69d4db#diff-c65eb52376398392f1395f8127364078L36), right? Do I understand correctly that leaving entity_id unassigned will make Home Assistant auto-generate an appropriate name?
There's no question about ```unique_id``` including the full identifier.

Reply:
Answering to my own question: yes. Even better, the patch will be to remove L36 altogether.